app/src/main/java/com/nervousfish/nervousfish/activities/SharedActivityMethods.java

```diff
@@ -0,0 +1,182 @@
+package com.nervousfish.nervousfish.activities;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.widget.EditText;
+
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.data_objects.IKey;
+import com.nervousfish.nervousfish.modules.database.IDatabase;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import cn.pedant.SweetAlert.SweetAlertDialog;
+
+/**
+ * Contains shared functionality between classes to reduce code duplication
+ */
+// Suppressed because we have an empty enum
+@SuppressWarnings("checkstyle:nowhitespacebefore")
+enum SharedActivityMethods {
```

Comment:
I think this enum should only be used for on contact receive, so maybe change its name to reflect that?